### PR TITLE
Fix Alpha Bugs

### PR DESCRIPTION
**Pull request by ethan-goh**
This PR addresses the alpha bugs related to Issues #138, #139, and #140. The fixes ensure a smoother user interface, accurate data updates in the display panel, and standardised text formatting.

### **Bug Fixes:**
**Bug 1: Black Bars Wrongly Shown in UI**

- **Issue:** Black bars appeared in unfilled areas of the `PersonListPanel` when there were fewer entries in the list, creating a visually inconsistent interface.

- **Solution:** Added explicit background color settings for empty cells within `PersonListPanel`. Now, any empty space in the list is consistently rendered with a white background.

- **Files Affected:**
  - `PersonListPanel.java`: Updated `updateItem` method in `PersonListViewCell` to ensure all cells, including empty ones, have a white background.

**Bug 2: Edited Person Details Not Reflected in Display**

- **Issue:** When editing a `Person` currently displayed in the details panel, changes were not immediately reflected.

- **Solution:** Modified the logic to update the display when an edited `Person` is the one currently being shown. This ensures real-time updates.

- **Files Affected:**
  - `EditCommand.java`: Added logic to refresh the details view with the edited `Person` if they are currently displayed.
  - `GradeCommand.java`: Added logic to refresh the details view with the edited `Person` if they are currently displayed.
  - `DeleteCommand.java`: Added logic to refresh the details view with the edited `Person` if they are currently displayed.
  - `ModuleCommand.java`: Added logic to refresh the details view with the edited `Person` if they are currently displayed.
  - `ModelManager.java`: Updated `setPersonToDisplay` to allow the display update upon edits.

**Bug 3: Standardise Capitalisation of Field Formats**

- **Issue:** Inconsistent capitalisation for user input fields, leading to varied formats across entries.

- **Solution:** Added methods in `ParserUtil` to handle capitalisation:
  - `makeCapitalise`: Capitalises the first letter of each word.
  - `makeUppercase` and `makeLowercase`: Convert strings to all uppercase or all lowercase, respectively.
 
- **Files Affected:**
  - `ParserUtil.java`: Added the new methods for capitalisation. Utilised new methods in the various parsing methods in this file.
  
### **Test Removal**: `execute_noFieldSpecifiedUnfilteredList_success`
**Reason for Deletion:** The `execute_noFieldSpecifiedUnfilteredList_success` test has been removed because the `EditCommand` is now designed to require at least one field to be specified for editing. Allowing the command to execute with no fields provided is not aligned with the intended functionality.

### CODE REVIEW

**Review by JYL27 — COMMENTED** [review]
Would it be better to handle the standardisation of the capitalisation of the fields in their respective classes instead?

File: src/main/java/seedu/address/logic/commands/ModuleCommand.java [hunk 1]
```diff
@@ -75,6 +75,12 @@ public CommandResult execute(Model model) throws CommandException {
         Person updatedPerson = person.addModule(module);
         model.setPerson(person, updatedPerson);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        System.out.println(updatedPerson);
+        System.out.println(model.getPersonToDisplay());
```
Comment (JYL27) [inline src/main/java/seedu/address/logic/commands/ModuleCommand.java:79]:
```suggestion
```

**Review by JYL27 — APPROVED** [review]
Print statements in ModuleCommand can be removed, otherwise LGTM!

**Comment by ethan-goh** [comment]
Remaining lines uncovered in codecov are in PersonDetails.java and PersonListPanel.java, which we are not unit testing

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -95,11 +96,29 @@ public CommandResult execute(Model model) throws CommandException {
                 throw new CommandException(String.format(MESSAGE_MODULE_NOT_FOUND, toDelete.getStudentId()));
             }
 
-            model.deleteModule(toDelete, moduleToDelete);
+            model.deleteModule(toDelete, module);
```
Comment (JYL27) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:99]:
```suggestion
            model.deleteModule(toDelete, moduleToDelete);
```

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -95,11 +96,29 @@ public CommandResult execute(Model model) throws CommandException {
...
+            if (toDelete.isSamePerson(model.getPersonToDisplay())) {
+                for (Person person : lastShownList) {
+                    if (person.getStudentId().equals(studentId)) {
+                        toDelete = person;
+                        break;
+                    }
+                }
+                model.setPersonToDisplay(toDelete);
+                return new CommandResult(String.format(MESSAGE_DELETE_MODULE_SUCCESS, module.toString()), true);
```
Comment (JYL27) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:108]:
```suggestion
                return new CommandResult(String.format(MESSAGE_DELETE_MODULE_SUCCESS, moduleToDelete), true);
```

**Review by JYL27 — COMMENTED** [review]
Some issues with the merge conflict resolution in DeleteCommand, otherwise LGTM

**Review by JYL27 — APPROVED** [review]
LGTM

**Pull request merged**